modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue

```diff
@@ -88,6 +127,86 @@ export default {
         }
       }
       return false;
+    },
+    compatible(title, slug) {
+      if ((typeof title) !== 'string') {
+        title = '';
+      }
+      if (this.page) {
+        const matches = slug.match(/[^\/]+$/);
+        slug = matches[0] || '';
+      }
+      return ((title === '') && (slug === `${this.prefix}none`)) || this.slugify(title) === this.slugify(slug);
+    },
+    slugify(s) {
+      const options = {};
+      if (this.field.page) {
+        options.allow = '/';
+      }
+      let preserveSlash = false;
+      // When you are typing a slug it feels wrong for hyphens you typed
+      // to disappear as you go, so if the last character is not valid in a slug,
+      // restore it after we call sluggo for the full string
+      if (this.focus && s.length && (sluggo(s.charAt(s.length - 1), options) === 'none')) {
+        preserveSlash = true;
+      }
+      s = sluggo(s, options);
+      if (preserveSlash) {
+        s += '-';
+      }
+      if (this.field.page) {
+        if (!s.charAt(0) !== '/') {
+          s = `/${s}`;
+        }
+        s = s.replace(/\/+/g, '/');
+        if (s !== '/') {
+          s = s.replace(/\/$/, '');
+        }
+      }
+      if (!s.length) {
+        s = 'none';
+      }
+      if (!s.startsWith(this.prefix)) {
+        if (this.prefix.startsWith(s)) {
+          // If they delete the `-`, and the prefix is `recipe-`,
+          // we want to restore `recipe-`, not set it to `recipe-recipe`
+          s = this.prefix;
+        } else {
+          s = this.prefix + s;
+        }
+      }
+      return s;
+    },
+    async checkConflict() {
+      let slug;
+      try {
+        slug = this.next;
+        await apos.http.post(`${apos.doc.action}/slug-taken`, {
+          body: {
+            slug,
+            _id: this.docId
+          }
+        });
+        // Still relevant?
+        if (slug === this.next) {
+          this.conflict = false;
+          this.validateAndEmit();
+        } else {
+          // Can ignore it
+        }
+      } catch (e) {
+        if (e.status === 409) {
```

Comment:
Can you add a bit more comment documentation here? Is this the error code if the slug is taken?

Reply:
Yes (409 Conflict).